sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala

```diff
@@ -93,6 +93,11 @@ object HiveThriftServer2 extends Logging {
       } else {
         None
       }
+      // If application was killed before HiveThriftServer2 start successfully then SparkSubmit
+      // process can not exit, so check whether if SparkContext was stopped.
+      if (SparkSQLEnv.sparkContext.stopped.get()) {
+        System.exit(-1)
```

Comment:
Since we have `logInfo("HiveThriftServer2 started")` at a few lines above, can we add a log at here to say that sparkcontext is stopped and we will exit?